<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AssignValuesWidget</class>
 <widget class="QWidget" name="AssignValuesWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>350</width>
    <height>403</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>350</width>
    <height>0</height>
   </size>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Assign Values</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="1">
    <layout class="QHBoxLayout" name="advancedLayout">
     <item alignment="Qt::AlignRight">
      <widget class="QPushButton" name="advancedButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Show advanced</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="7" column="1">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Select Assignment Method</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_5">
      <item row="1" column="0">
       <widget class="QStackedWidget" name="stackedWidget">
        <property name="currentIndex">
         <number>2</number>
        </property>
        <widget class="QWidget" name="page">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>115</height>
          </size>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <layout class="QFormLayout" name="formLayout">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="IDlabelStartValue">
              <property name="text">
               <string>Start ID</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBoxIDStartValue">
              <property name="minimum">
               <number>-100000</number>
              </property>
              <property name="maximum">
               <number>100000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="0">
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="0">
           <layout class="QFormLayout" name="formLayout_2">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="LabelInitialString">
              <property name="text">
               <string>Initial string</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="LineEditInitialString">
              <property name="text">
               <string>a</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_5">
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="0">
           <layout class="QFormLayout" name="formLayout_5">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="LabelAlphaNumericID">
              <property name="text">
               <string>Start ID</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="SpinBoxAlphaNumericIDStart">
              <property name="minimum">
               <number>-100000</number>
              </property>
              <property name="value">
               <number>0</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="LabelAlphaNumericPrefix">
              <property name="text">
               <string>Prefix</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="LineEditAlphaNumericPrefix"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_3">
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <layout class="QFormLayout" name="formLayout_3">
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::ExpandingFieldsGrow</enum>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Lower Limit</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinBoxIntegerLowerLimit">
              <property name="minimum">
               <number>-99999</number>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Upper Limit</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Generator Seed</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="spinBoxIntegerGeneratorSeed">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spinBoxIntegerUpperLimit">
              <property name="minimum">
               <number>-99999</number>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
              <property name="value">
               <number>100</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_4">
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <layout class="QFormLayout" name="formLayout_4">
            <item row="0" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Lower Limit</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Upper Limit</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Generator Seed</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QDoubleSpinBox" name="spinBoxFloatLowerLimit">
              <property name="decimals">
               <number>3</number>
              </property>
              <property name="minimum">
               <double>-99999.990000000005239</double>
              </property>
              <property name="value">
               <double>0.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="spinBoxFloatUpperLimit">
              <property name="decimals">
               <number>3</number>
              </property>
              <property name="minimum">
               <double>-99999.990000000005239</double>
              </property>
              <property name="maximum">
               <double>99999.990000000005239</double>
              </property>
              <property name="value">
               <double>10.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="spinBoxFloatGeneratorSeed">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>99999</number>
              </property>
              <property name="value">
               <number>1</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="KComboBox" name="comboBoxMethod">
        <item>
         <property name="text">
          <string>Enumerate Items</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Enumerate Alpha</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Enumerate Alpha-Numeric</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Assign Random Integers</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Assign Random Reals</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <layout class="QGridLayout" name="groupOverwrite">
        <item row="3" column="1">
         <widget class="QCheckBox" name="checkBoxOverwriteValues">
          <property name="text">
           <string/>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Overwrite existing values</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QGroupBox" name="groupTarget">
     <property name="baseSize">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
     <property name="title">
      <string>Select Target Elements</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_7">
      <item row="2" column="0">
       <widget class="QLabel" name="labelData">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Data</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="KLineEdit" name="propertyName"/>
      </item>
      <item row="0" column="2">
       <widget class="KComboBox" name="dataStructuresCombo">
        <property name="toolTip">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'DejaVu Sans'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;List of all data structures that are of type &lt;span style=&quot; font-style:italic;&quot;&gt;Graph&lt;/span&gt;.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="labelDataStructure">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Data Structure</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelProperty">
        <property name="text">
         <string>Property Name</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelConnection">
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Connection</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="KComboBox" name="connectionType">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="applyToConnections">
          <property name="toolTip">
           <string>Assign values to selected connections.</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="2">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="KComboBox" name="dataType">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="applyToDataElements">
          <property name="toolTip">
           <string>Assign values to selected data elements.</string>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>comboBoxMethod</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>141</y>
    </hint>
    <hint type="destinationlabel">
     <x>308</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>checkBoxOverwriteValues</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>253</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>label</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>96</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBoxIntegerGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>124</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_7</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>spinBoxFloatGeneratorSeed</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>69</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>label_9</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>323</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>LabelInitialString</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>52</x>
     <y>263</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>advancedButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>LineEditInitialString</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>299</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>200</x>
     <y>267</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
